### DISCUSSION TITLE
Where to find body html tag to put some scripts in it at index.md or index.mdx file of docusaurus?

### DISCUSSION THREAD

**Discussion by ashishkharche** [facebook/docusaurus#6639] [Q&A]
A question regarding adding netlifyCMS to docusaurus website.

In a tutorial, I am stuck at this process where it requires me to add a script in Docusaurus `Head` and `Body` but the file is mdx and I am not sure how to make it happen?

Quoting: "To do this, add the following script before the closing body tag of your site's main index page:"

```js
<script>
  if (window.netlifyIdentity) {
    window.netlifyIdentity.on("init", user => {
      if (!user) {
        window.netlifyIdentity.on("login", () => {
          document.location.href = "/admin/";
        });
      }
    });
  }
</script>
```

Link to the article: https://www.netlifycms.org/docs/add-to-your-site/#add-the-netlify-identity-widget

Basically as the `.mdx` or `.md` index file of my docusaurus code is not HTML, I am not sure where to find the `body` tag and put that script there.

**Comment by Josh-Cena** [comment]
Responded on Discord.

**Reply by joe-miskolczy** [reply]
What was the answer? I would love to do the same but couldn't find a good solution so far.

**Reply by Josh-Cena** [reply]
@JOE-SNWPLW Put this into a separate client module. See https://docusaurus.io/docs/advanced/client#client-modules.

Although I believe authentication would be somewhat more complicated than that...

**Reply by ashishkharche** [reply]
@JOE-SNWPLW 

In `src/pages/index.js`

```js
/*
{
"title": "Index page of the site",
  "description": "This is the Index page or home page of the site. "
}
*/
import React from "react";
import Layout from "@theme/Layout";
import Head from '@docusaurus/Head';
import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
import {Community, Intro,} from "../components";

export default function Home() {
    const context = useDocusaurusContext();
    const {siteConfig = {}} = context;

    // Add Netlify Identity script
    if (typeof window !== "undefined") {
        // browser code
        const document = window.document;
        let script = document.createElement("script");
        script.innerHTML = `
      if (window.netlifyIdentity) {
        window.netlifyIdentity.on("init", user => {
          if (!user) {
            window.netlifyIdentity.on("login", () => {
              document.location.href = "/admin/";
            });
          }
        });
      }
    `;
        document.head.appendChild(script);
    }

    return (
        <div className="main">
            <Layout
                className="my-2 mx-auto w-full max-w-screen-lg px-8 shadow-none"
                title={`${siteConfig.title}`}
                description={`${siteConfig.tagline}`}
            >
                <Head>
                    <script src="https://identity.netlify.com/v1/netlify-identity-widget.js"></script>
                </Head>
                <main className="mx-auto max-w-screen-lg p-6 md:p-10">
                    <Intro/>
                    <Community/>
                </main>
            </Layout>
        </div>
    );
}

```

In `static/admin/index.html`

```html
<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Content Manager</title>

  <!-- Include the script for Netlify Identity Widget -->
  <script src="https://identity.netlify.com/v1/netlify-identity-widget.js"></script>
</head>

<body>
  <!-- Include the script that builds the page and powers Netlify CMS -->
  <script src="https://unpkg.com/netlify-cms@^2.10.192/dist/netlify-cms.js"></script>
</body>

</html>
```

**Reply by slorber** [reply]
@ashishkharche rendering a component should not produce side effects, your solution is not ideal and would only add the script to the DOM while React tries to render.

Our plugin API allows you to add a post body tag so that your inline script appears in the static HTML files directly (not needing to execute any js to make it appear here)

https://docusaurus.io/docs/api/plugin-methods/lifecycle-apis#injectHtmlTags

**Reply by joe-miskolczy** [reply]
@Josh-Cena @ashishkharche @slorber Thanks for all the suggestions, they helped to get on the right track!

**Reply by harunisik** [reply]
injectHtmlTags appends the new provided tags only. Is there any way to manipulate the tags added by docusaurus?

For example, I want to completely remove this line from the generated HTML;

`<link rel="alternate" href="https://mywebsite.com/" hreflang="x-default" data-rh="true">`

**Reply by slorber** [reply]
@harunisik this is set in the theme and not in Docusaurus core, so hopefully you can still swizzle --eject SiteMetadata to remove it.

Now it would be nice to help us understand why you want to remove this line in the first place. We try to have good SEO defaults. Please report here: https://github.com/facebook/docusaurus/issues/6075

**Reply by harunisik** [reply]
@slorber Thanks for your suggestion. Swizzling is the better option for me for now. We're trying to fix the issues reported by our website audit checks. I'll also report those problems under #6075.

**Reply by slorber** [reply]
Thanks @harunisik , please do report.

Afaik audit tools may report bad info sometimes and do not agree with each others. In my case Ahrefs for example reported bad errors and their support couldn't really explain me what I did wrong on Docusaurus. Related to https://github.com/facebook/docusaurus/issues/6011#issuecomment-994718607

That's not really related to this discussion so please report your audit issue on the appropriate issue here https://github.com/facebook/docusaurus/issues/6075
